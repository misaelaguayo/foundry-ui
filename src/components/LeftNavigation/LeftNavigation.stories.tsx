import React from 'react';
import { Story, Meta } from '@storybook/react';
import styled from 'styled-components';
import Icon from '@mdi/react';
import * as Icons from '@mdi/js';
import LeftNavigation, { LeftNavigationProps } from './LeftNavigation';
import Button from '../Button';

type DefaultProps = LeftNavigationProps;

const getIconPath = (path: string, color: string) =>
  path ? <Icon color={color} size="1rem" path={path} /> : undefined;

const App = styled.div`
`;

const Ipsum = styled.div`
    padding-left: 250px;
    top: 0;
`;

const lorem = `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
ut labore et dolore magna aliqua. Porttitor massa id neque aliquam vestibulum morbi
blandit cursus risus. Mi bibendum neque egestas congue quisque egestas diam. Pulvinar
sapien et ligula ullamcorper. Ac turpis egestas integer eget aliquet nibh. Bibendum at
varius vel pharetra vel turpis nunc eget. Interdum varius sit amet mattis vulputate enim
nulla aliquet porttitor. Morbi tempus iaculis urna id volutpat. Auctor neque vitae tempus
quam pellentesque. Tempor commodo ullamcorper a lacus vestibulum sed. Malesuada proin
libero nunc consequat interdum varius. At volutpat diam ut venenatis tellus in metus
vulputate eu. Quam adipiscing vitae proin sagittis nisl. Sagittis purus sit amet volutpat
consequat. Magna fringilla urna porttitor rhoncus dolor. Vitae aliquet nec ullamcorper sit
amet. Imperdiet dui accumsan sit amet nulla facilisi. Sapien faucibus et molestie ac
feugiat sed lectus vestibulum. Sed tempus urna et pharetra pharetra. Ut placerat orci
nulla pellentesque dignissim enim sit amet. Netus et malesuada fames ac turpis egestas
integer. Bibendum est ultricies integer quis auctor elit sed. Et netus et malesuada fames
ac turpis. Ultrices sagittis orci a scelerisque purus semper eget duis at. Non blandit
massa enim nec dui nunc. Felis bibendum ut tristique et egestas quis. Volutpat ac
tincidunt vitae semper. Adipiscing vitae proin sagittis nisl rhoncus mattis rhoncus urna.
Donec enim diam vulputate ut pharetra sit amet aliquam. Quis enim lobortis scelerisque
fermentum dui faucibus in ornare quam. Suspendisse sed nisi lacus sed viverra tellus in.
Proin nibh nisl condimentum id venenatis. Duis tristique sollicitudin nibh sit amet
commodo nulla. Sed cras ornare arcu dui. Consectetur adipiscing elit pellentesque habitant
morbi tristique senectus et netus. Ut etiam sit amet nisl purus in mollis nunc. Aliquam id
diam maecenas ultricies mi eget mauris pharetra et. Neque volutpat ac tincidunt vitae
semper quis lectus. Amet nisl suscipit adipiscing bibendum. Ultrices sagittis orci a
scelerisque purus. Et malesuada fames ac turpis egestas sed tempus. Euismod lacinia at
quis risus sed vulputate. Egestas fringilla phasellus faucibus scelerisque eleifend donec.
Nulla posuere sollicitudin aliquam ultrices sagittis orci a scelerisque purus. Nunc
lobortis mattis aliquam faucibus purus in massa tempor. Euismod nisi porta lorem mollis
aliquam. Varius vel pharetra vel turpis nunc eget lorem dolor sed. Tempus urna et pharetra
pharetra massa massa ultricies mi quis. Aliquam faucibus purus in massa. Pellentesque sit
amet porttitor eget dolor. Sit amet cursus sit amet dictum sit amet. Adipiscing elit ut
aliquam purus sit. Velit ut tortor pretium viverra suspendisse. Aliquet nibh praesent
tristique magna sit amet purus gravida. Risus pretium quam vulputate dignissim suspendisse
in est ante. Nulla pharetra diam sit amet nisl suscipit adipiscing bibendum est. Tristique
et egestas quis ipsum suspendisse ultrices gravida. Aliquam ut porttitor leo a diam. Quis
hendrerit dolor magna eget est lorem. Sed tempus urna et pharetra pharetra massa massa
ultricies mi. Venenatis urna cursus eget nunc scelerisque viverra mauris. Sed egestas
egestas fringilla phasellus faucibus scelerisque eleifend. Venenatis tellus in metus
vulputate eu. Parturient montes nascetur ridiculus mus. Id diam vel quam elementum
pulvinar etiam. Magna etiam tempor orci eu.`;

const CloseButton = styled(Button.Container)`
  top: 0;
  right: 0;
  padding: 0.5rem;
  position:fixed;
`;

export const Default: Story<DefaultProps> = ({ ...args }: DefaultProps) => {
    const [hidden, setHidden] = React.useState(args.hidden);
    React.useEffect(() => setHidden(args.hidden), [args.hidden]);
    return (
        <App>
            <LeftNavigation
              {...args}
              HrefColor={args.HrefColor}
              bgcolor={args.bgcolor}
              hidden={hidden}
              header={
                <Button
                  color="#0000"
                  StyledContainer={CloseButton}
                  onClick={() => setHidden(!hidden)}
                >
                  {getIconPath(Icons.mdiClose, 'white')}
                </Button>
              }
            />
            <Ipsum>
                <h2>Lorem Ipsum</h2>
                <p>{lorem}</p>
            </Ipsum>
        </App>

    );
};

Default.args = {
    navButtons: [
      {
            label: 'Sample Link 1',
            link: 'route1',
            onClick: () => {},
        },
      {
            label: 'Sample Link 2',
            link: 'route1',
            onClick: () => {},
        },
      {
            label: 'Sample Link 3',
            link: 'route1',
            onClick: () => {},
        },
      {
            label: 'Sample Link 4',
            link: 'route1',
            onClick: () => {},
        },
        {
            label: 'Sample Link 5',
            link: 'route2',
            onClick: () => {},
        }],
    footer: 'Sample footer',
    hidden: false,
    width: '200px',
    bgcolor: '#BF3A00',
  };

export default {
    title: 'LeftNavigation',
    component: LeftNavigation,
    argTypes: {
        position: {
            options: ['relative', 'sticky', 'fixed', 'static'],
            control: {
                type: 'select',
            },
        },
    },
    parameters: {},
} as Meta;
