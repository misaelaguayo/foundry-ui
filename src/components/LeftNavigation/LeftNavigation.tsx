import React from 'react';
import styled from 'styled-components';
import { StyledSubcomponentType, SubcomponentPropsType } from '../commonTypes';
import { Div, A } from '../../htmlElements';

export const Container = styled(Div)`
${({
    bgcolor,
    animation,
  }: {
    bgcolor: string;
    animation: () => void;
  }) => `
  padding-top: 20px;
  height: 100%;
  width: 250px;
  position: fixed;
  color: white;
  background-color: ${bgcolor};
  top: 0;
  ${animation}
  `}
`;

export type HideAnimationPropType = {
    length?: number;
    origin?: string;
    collapsed?: boolean;
};

export const defaultHideAnimation = ({
    length = 0.1,
    origin = 'left',
    collapsed,
}: HideAnimationPropType) => `
    transform: ${collapsed ? 'scaleX(0)' : 'scaleX(1)'};
    transform-origin: ${origin};
    transition: transform ${length}s cubic-bezier(0, .7, .9, 1);;
`;

export const ContainerHref = styled(A)`
${({
    HrefColor,
} : {
    HrefColor: string;
}) => `
    padding: 8px 8px 8px 32px;
    text-decoration: none;
    font-size: 25px;
    color: ${HrefColor};
    display: block;
    `}
`;

export type NavButton = {
    label: string;
    link: string;
    onClick: () => void;
}

export interface LeftNavigationProps {
    StyledContainer?: StyledSubcomponentType,
    navButtons?: NavButton[];
    color?: string;
    bgcolor?: string;
    HrefColor?: string;
    containerProps?: SubcomponentPropsType;
    hidden?: boolean;
    hideAnimation?: (value: HideAnimationPropType) => void;
}

const LeftNavigation = ({
    StyledContainer = Container,
    navButtons,
    bgcolor = 'Black',
    HrefColor = 'White',
    containerProps,
    hidden = false,
    hideAnimation = defaultHideAnimation,
}: LeftNavigationProps): JSX.Element => {
    const [isHidden] = React.useState(false);
    const animationProps = { collapsed: hidden || isHidden };
    return (
        <StyledContainer
          bgcolor={bgcolor}
          animation={hideAnimation(animationProps)}
          {...containerProps}
        >
            {navButtons &&
                navButtons.map((navButton) => (
                    <ContainerHref HrefColor={HrefColor} href={navButton.link}>{navButton.label}</ContainerHref>
                ))}
        </StyledContainer>
    );
};

export default LeftNavigation;
