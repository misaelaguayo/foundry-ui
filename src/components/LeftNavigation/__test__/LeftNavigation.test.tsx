import { toHaveNoViolations } from 'jest-axe';
import { render, waitFor, configure } from '@testing-library/react';
import LeftNavigation from '../LeftNavigation';
import { axe } from 'jest-axe';
import React from 'react'

expect.extend(toHaveNoViolations)

configure({testIdAttribute: 'data-test-id'});
const testId = 'foundry-leftnavigation'

describe('LeftNavigation', () => {
    describe('Accessibility Tests', () => {
        it('Should pass accessibility test with default props', async () => {
            const component = <LeftNavigation bgcolor="Black" />;
            const { container } = render(component);
            const results = await axe(container);
            expect(results).toHaveNoViolations();
        });
    });

    it('shows LeftNavigation with default props', async () => {
        const { container, getByTestId } = render(
            <LeftNavigation containerProps={{'data-test-id': testId}}/>,
        );

        await waitFor(() => getByTestId(testId));
        expect(container).toMatchSnapshot();
    });

    it('shows LeftNavigation with header props', async() => {
        const { container, getByTestId } = render(
            <LeftNavigation header="testing header" headerProps={{'data-test-id': testId}} />
        );

        await waitFor(() => getByTestId(testId));
        expect(container).toMatchSnapshot();
    });

    it('shows LeftNavigation with footer props', async() => {
        const { container, getByTestId } = render(
            <LeftNavigation footer="testing footer" footerProps={{'data-test-id': testId}} />
        );

        await waitFor(() => getByTestId(testId));
        expect(container).toMatchSnapshot();
    });

});