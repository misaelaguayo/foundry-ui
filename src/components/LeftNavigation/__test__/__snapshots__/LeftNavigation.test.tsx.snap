// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftNavigation shows LeftNavigation with default props 1`] = `
.c0 {
  padding-top: 20px;
  height: 100%;
  width: undefined;
  position: fixed;
  color: white;
  background-color: Black;
  top: 0;
  -webkit-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
  -webkit-transform-origin: left;
  -ms-transform-origin: left;
  transform-origin: left;
  -webkit-transition: -webkit-transform 0.1s cubic-bezier(0,.7,.9,1);
  -webkit-transition: transform 0.1s cubic-bezier(0,.7,.9,1);
  transition: transform 0.1s cubic-bezier(0,.7,.9,1);
}

.c1 {
  padding: 1.5rem 1.5rem;
  color: #51575D;
}

<div>
  <div
    class=" c0"
    data-test-id="foundry-leftnavigation"
  >
    <div
      class=" c1"
    />
  </div>
</div>
`;

exports[`LeftNavigation shows LeftNavigation with footer props 1`] = `
.c0 {
  padding-top: 20px;
  height: 100%;
  width: undefined;
  position: fixed;
  color: white;
  background-color: Black;
  top: 0;
  -webkit-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
  -webkit-transform-origin: left;
  -ms-transform-origin: left;
  transform-origin: left;
  -webkit-transition: -webkit-transform 0.1s cubic-bezier(0,.7,.9,1);
  -webkit-transition: transform 0.1s cubic-bezier(0,.7,.9,1);
  transition: transform 0.1s cubic-bezier(0,.7,.9,1);
}

.c1 {
  padding: 1.5rem 1.5rem;
  color: #51575D;
}

.c2 {
  padding: 10rem 1.5rem;
  color: #D5D5D5;
  border-radius: 0rem 0rem 0.25rem 0.25rem;
}

<div>
  <div
    class=" c0"
  >
    <div
      class=" c1"
    />
    <div
      class=" c2"
      data-test-id="foundry-leftnavigation"
    >
      testing footer
    </div>
  </div>
</div>
`;

exports[`LeftNavigation shows LeftNavigation with header props 1`] = `
.c0 {
  padding-top: 20px;
  height: 100%;
  width: undefined;
  position: fixed;
  color: white;
  background-color: Black;
  top: 0;
  -webkit-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
  -webkit-transform-origin: left;
  -ms-transform-origin: left;
  transform-origin: left;
  -webkit-transition: -webkit-transform 0.1s cubic-bezier(0,.7,.9,1);
  -webkit-transition: transform 0.1s cubic-bezier(0,.7,.9,1);
  transition: transform 0.1s cubic-bezier(0,.7,.9,1);
}

.c1 {
  padding: 1.5rem 1.5rem 0rem border-radius:0.25rem 0.25rem 0rem 0rem;
  font-weight: bold;
  color: #252D34;
}

.c2 {
  padding: 1.5rem 1.5rem;
  color: #51575D;
}

<div>
  <div
    class=" c0"
  >
    <div
      class=" c1"
      data-test-id="foundry-leftnavigation"
    >
      testing header
    </div>
    <div
      class=" c2"
    />
  </div>
</div>
`;
